File: compiler/testData/foreignAnnotationsJava8/tests/jspecify_tests/Simple.java
import org.jspecify.annotations.*;

@DefaultNotNull
public class Simple {
    // PSometimesNull(_)
    @Nullable public String field = null;

    // PSometimesNull(_), PNonNull(x), PUnknownNull(y)
    @Nullable
    public String foo(String x, @NullnessUnknown CharSequence y) {
        return "";
    }

    // PNonNull(_)
    @NotNull
    public String bar() {
        return "";
    }
}

File: compiler/testData/foreignAnnotationsJava8/tests/jspecify/warnings/simple.kt
// !DIAGNOSTICS: -UNUSED_VARIABLE -UNUSED_PARAMETER
// CODE_ANALYSIS_STATE warn
// FILE: A.java

import org.jspecify.annotations.*;

public class A {
    @Nullable public String field = null;

    @Nullable
    public String foo(String x, @NullnessUnknown CharSequence y) {
        return "";
    }
}

// FILE: main.kt

fun main(a: A) {
    a.foo("", null)?.length
    <!RECEIVER_NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS!>a.foo("", null)<!>.length

    a.field?.length
    <!RECEIVER_NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS!>a.field<!>.length
}
